pkg/virt-controller/watch/vmi.go

```diff
@@ -2399,3 +2401,109 @@ func (c *VMIController) syncMemoryHotplug(vmi *virtv1.VirtualMachineInstance) {
 		vmi.Labels[virtv1.MemoryHotplugOverheadRatioLabel] = *overheadRatio
 	}
 }
+
+type aggregateDataVolumeCondition struct {
+	Status  k8sv1.ConditionStatus
+	Reason  []string
+	Message []string
+}
+
+func (c *VMIController) aggregateDataVolumesConditions(vmiCopy *virtv1.VirtualMachineInstance, dvs []*cdiv1.DataVolume) {
+	if len(dvs) == 0 {
+		return
+	}
+	const ConditionUninitialized k8sv1.ConditionStatus = "Uninitialized"
+
+	allDVConditions := map[cdiv1.DataVolumeConditionType]aggregateDataVolumeCondition{
+		cdiv1.DataVolumeReady: {
+			Status: ConditionUninitialized,
+		},
+		cdiv1.DataVolumeBound: {
+			Status: ConditionUninitialized,
+		},
+		cdiv1.DataVolumeRunning: {
+			Status: ConditionUninitialized,
+		},
+	}
+
+	for _, dv := range dvs {
+		for _, condition := range dv.Status.Conditions {
+			if allDVConditions[condition.Type].Status == ConditionUninitialized {
```

Comment:
Can't you do something like this instead of introducing the `ConditionUninitialized` const?

```
if _, ok := allDVConditions[condition.Type]; !ok {
	allDVConditions[condition.Type] = aggregateDataVolumeCondition{
		Status:  condition.Status,
		Reason:  []string{condition.Reason},
		Message: []string{condition.Message},
	}
} else {
	if condition.Type == cdiv1.DataVolumeRunning {
		aggregateCondsTrueWins(condition, allDVConditions)
	} else {
		aggregateCondsFalseWins(condition, allDVConditions)
	}
}
```

Reply:
This should work with your status checks below too, because the returned default value should still satisfy the checks (default `Status` is empty).

Reply:
Right, I can simply consider an empty status as uninitialized.
Done